uefi/src/proto/device_path/mod.rs

```diff
@@ -184,6 +185,21 @@ impl DevicePathNode {
         &self.data
     }
 
+    /// Returns the payload data of this node as u16 slice,
+    /// if the slice can be mapped to a `u16` slice..
+    #[must_use]
+    pub fn data_u16(&self) -> Option<&[u16]> {
```

Comment:
I decided for this approach, as having a CStr16::from_u8_slice_with_nul has the problem that there is no sane way of guaranteeing `[u8]` to be properly aligned. miri complains

Reply:
Accessing node data is usually best done through the generated types: https://docs.rs/uefi/latest/uefi/proto/device_path/index.html#modules

For example, https://docs.rs/uefi/latest/uefi/proto/device_path/media/struct.FilePath.html provides direct access to the path string within a media file path node.